File: src/main/java/rs/ac/uns/ftn/Bookify/model/ReportedUser.java
package rs.ac.uns.ftn.Bookify.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "reported_users")
public class ReportedUser {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(nullable = false)
	private String reason;

	@Column(nullable = false)
	private Date created;

	@ManyToOne
	private User reportedUser;

	@ManyToOne
	private User createdBy;

}

File: Bookify/bookify-client/Bookify/src/app/feature-modules/administration/model/reported.user.ts
export interface ReportedUser {
    id?:number,
    reason?:string,
    created?:number,
    reportedUser:{
        uid?:string,
        firstName?:string,
        lastName?:string,
        email?:string,
        blocked: boolean
    },
    createdBy:{
        uid?:string,
        firstName?:string,
        lastName?:string,
        email?:string
    }

}
